### DISCUSSION TITLE
Error parsing mesh ("Bad integer")

### DISCUSSION THREAD

**Discussion by Yuanguo1994** [neperfepx/FEPX#5] [Q&A] [answered]
Hi, I got a confusing error and want to get some help. When I use the simulation.msh build by myself, sometimes the error will occur as follows:

'At line 1362 of file /mnt/c/Ubuntu-app/fepx_app/fepx-1.1.1/src/libfepx/read_input_mod.f90 (unit = 73, file = 'simulation.msh')
Fortran runtime error: Bad integer for item 1 in list input'

I want to know what does this mean? BTW, I have checked the line 1362 as follows:
'1362 2 3 3 3 0 842 813 854'

**Comment by jcappola** [comment, marked as answer]
When you say "build yourself" do you mean with Neper or by hand? That error suggests that at line 1362 of `read_input_mod.f90` there is an error processing something in your mesh file (not at line 1362 of _your_ mesh file).

Could you please provide the mesh file (change the file suffix from `.msh` to `.txt`) so we can figure it out?

**Reply by Yuanguo1994** [reply]
> When you say "build yourself" do you mean with Neper or by hand? That error suggests that at line 1362 of `read_input_mod.f90` there is an error processing something in your mesh file (not at line 1362 of _your_ mesh file).
> 
> Could you please provide the mesh file (change the file suffix from `.msh` to `.txt`) so we can figure it out?

Hi, Thanks for your reply. I built the mesh with Neper. 

I have attached the file as follows.
[simulation.txt](https://github.com/acmelab-ua/FEPX/files/6097451/simulation.txt)

**Reply by acmelab-ua** [reply]
Please also provide the Neper commands you used to build the tessellation and mesh. This could very well be an issue with how you are using Neper, and not an issue with FEPX.

**Reply by jcappola** [reply]
A quick glace tells me that the 3D elements are _likely_ linear tetrahedral elements (4 nodes). When building a mesh for FEPX via Neper you must include an `-order 2` flag in your call to the meshing module (`-M`).

**Reply by Yuanguo1994** [reply]
> A quick glace tells me that the 3D elements are _likely_ linear tetrahedral elements (4 nodes). When building a mesh for FEPX via Neper you must include an `-order 2` flag in your call to the meshing module (`-M`).

yes, i didnt include 'order 2' when mesh the tess. Now it works. Thank!

**Reply by acmelab-ua** [reply]
Please note that the example simulations provided with FEPX contain scripts named `generate_mesh.sh`, which contain Neper commands to make a proper tessellation and mesh for FEPX. Not all portions of these commands are necessary to make the input for FEPX, but they should give you a sense of where to start with polycrystal and mesh generation. You can consult the Neper manual to explore what each of the aspects of the commands do, which will help illuminate the portions that are necessary for FEPX.

**Reply by rquey** [reply]
@acmelab-ua, This will likely be a very common user issue... Would it be possible for FEPX to print a clear error message when it occurs?
